multiview/cluster/mv_spectral.py

```diff
@@ -0,0 +1,232 @@
+
+        Returns
+        -------
+        la_eigs : array-like, shape(n_samples, n_clusters)
+            The top n_cluster eigenvectors of the normalized graph
+            laplacian.
+        '''
+
+        # Compute the normalized laplacian
+        d_mat = np.diag(np.sum(X, axis=1))
+        d_alt = np.sqrt(np.linalg.inv(np.abs(d_mat)))
+        laplacian = d_alt @ X @ d_alt
+
+        # Make the resulting matrix symmetric
+        laplacian = (laplacian + np.transpose(laplacian)) / 2.0
+
+        # Obtain the top n_cluster eigenvectors of the laplacian
+        e_vals, e_vecs = np.linalg.eig(laplacian)
+        indices = np.argsort(np.real(e_vals))[-self.n_clusters:]
+        la_eigs = np.real(e_vecs[:, indices])
+
+        return la_eigs
+
+    def fit_predict(self, Xs):
+
+        '''
+        Performs clustering on the multiple views of data and returns
+        the cluster labels.
+
+        Parameters
+        ----------
+        Xs : list of array_likes
+            - Xs shape: (n_views,)
+            - Xs[0] shape: (n_samples, n_features_i)
+            This list must be of size n_views, corresponding to the number
+            of views of data. Each view can have a different number of
+            features, but they must have the same number of samples.
+
+        Returns
+        -------
+        predictions : array_like, shape(n_samples,)
+            The predicted cluster labels for each sample.
+        '''
+
+        Xs = check_Xs(Xs)
+        if len(Xs) != self.n_views:
+            msg = 'Length of Xs must be the same as n_views'
+            raise ValueError(msg)
+
+        # Compute the similarity matrices
+        sims = [self._gaussian_sim(X) for X in Xs]
+
+        # Initialize matrices of eigenvectors
+        U_mats = [self._compute_eigs(sim) for sim in sims]
+
+        # Iteratively compute new graph similarities, laplacians,
+        # and eigenvectors
+        for iter in range(self.max_iter):
+
+            # Compute the sums of the products of the spectral embeddings
+            # and their transposes
+            eig_sums = [u_mat @ np.transpose(u_mat) for u_mat in U_mats]
+            U_sum = np.sum(np.array(eig_sums), axis=0)
+            new_sims = list()
+
+            for view in range(self.n_views):
+                # Compute new graph similarity representation
+                mat1 = sims[view] @ (U_sum - eig_sums[view])
+                mat1 = (mat1 + np.transpose(mat1)) / 2.0
+                new_sims.append(mat1)
+                # Recompute eigenvectors
+                U_mats = [self._compute_eigs(sim)
+                          for sim in new_sims]
+
+        # Row normalize
+        for view in range(self.n_views):
+            U_norm = np.linalg.norm(U_mats[view], axis=1).reshape((-1, 1))
+            U_norm[U_norm == 0] = 1
+            U_mats[view] /= U_norm
+
+        # Performing kmeans clustering
+        kmeans = KMeans(n_clusters=self.n_clusters,
```

Comment:
this is another thing that doesn't necessarily need to be `KMeans` in the future. 

also there are more params one could pass down to `KMeans` - for example, I almost always increase `n_init` from the default and find that it really helps. Consider allowing parameters to be passed down to `KMeans` or even pass down a different clustering algorithm